antarest/common/config.py

```diff
@@ -1,33 +1,50 @@
 import os
 from copy import deepcopy
+from dataclasses import dataclass
 from pathlib import Path
-from typing import Any, Optional
+from typing import Any, Optional, Dict, Type, TypeVar, Callable, cast
 
 import yaml
+from dataclasses_json import DataClassJsonMixin  # type: ignore
+from dataclasses_json.api import dataclass_json
 
 from antarest.common.custom_types import JSON
 
+config_definitions: Dict[str, Type[DataClassJsonMixin]] = {}
```

Comment:
What is its purpose ?

Reply:
this is the config module typed config definitions (they are module scoped)